src/components/LocationAutocomplete/__tests__/LocationAutocomplete.spec.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { render, screen, RenderResult } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { stabGoogleApi, getPlacePredictionsMock } from '../../../__mocks__/googleApiMock';
+import { LocationAutocomplete } from '../LocationAutocomplete';
+import predictionsMock from '../__mocks__/predictions.json';
+
+stabGoogleApi();
+
+describe('<LocationAutocomplete/> that renders a location search field', () => {
+    let view: RenderResult;
+    const onSelectionMock = jest.fn();
+    const onErrorMock = jest.fn();
+    const onRemoveAllLocationsMock = jest.fn();
+
+    const focusField = async (user) => {
+        await user.click(screen.getByRole('textbox'));
+    };
+
+    beforeEach(() => {
+        view = render(
+            <LocationAutocomplete
+                inputPlaceholder="Location..."
+                noSuggestionsPlaceholder="No suggestions..."
+                onSelectionChange={onSelectionMock}
+                onError={onErrorMock}
+                onRemoveAllLocations={onRemoveAllLocationsMock}
+            />
+        );
+    });
+
+    it('should render with minimal props', () => {
+        expect(view.container).toMatchSnapshot();
+    });
+
+    it('should set loading to true when user starts typing', async () => {
+        const user = userEvent.setup();
+        const textBox = screen.getByRole('textbox');
+        await user.type(textBox, 'Honolulu');
+
+        expect(textBox).toHaveAttribute('value', 'Honolulu');
+    });
+
+    it('should set loading to false when user deletes input value', async () => {
+        const user = userEvent.setup();
+        const textBox = screen.getByRole('textbox');
+        await user.type(textBox, 'Honolulu');
+
+        expect(textBox).toHaveAttribute('value', 'Honolulu');
+
+        await user.type(textBox, ' ');
+    });
+
+    it('should request predictions from API when user types', async () => {
+        const user = userEvent.setup();
+        await user.type(screen.getByRole('textbox'), 'Tonga');
+        const textBox = screen.getByRole('textbox');
+
+        await focusField(user);
+
+        expect(textBox).toHaveAttribute('value', 'Tonga');
+    });
+
+    it('should pass predictions to AutosuggestDeprecated', async () => {
```

Comment:
this test description is not useful. It checks for internal logic, not user value. I'm not sure what the point of this test was. Maybe it can be deleted.

Reply:
:+1: